pkg/apis/pipeline/v1beta1/pipelinerun_conversion.go

```diff
@@ -208,6 +215,160 @@ func (ptrs *PipelineTaskRunSpec) convertFrom(ctx context.Context, source v1.Pipe
 	ptrs.ComputeResources = source.ComputeResources
 }
 
+// ConvertTo implements apis.Convertible
+func (prs *PipelineRunStatus) ConvertTo(ctx context.Context, sink *v1.PipelineRunStatus) error {
+	sink.Status = prs.Status
+	sink.StartTime = prs.StartTime
+	sink.CompletionTime = prs.CompletionTime
+	sink.Results = nil
+	for _, pr := range prs.PipelineResults {
+		new := v1.PipelineRunResult{}
+		pr.convertTo(ctx, &new)
+		sink.Results = append(sink.Results, new)
+	}
+	if prs.PipelineSpec != nil {
+		sink.PipelineSpec = &v1.PipelineSpec{}
+		err := prs.PipelineSpec.ConvertTo(ctx, sink.PipelineSpec)
+		if err != nil {
+			return err
+		}
+	}
+	sink.SkippedTasks = nil
+	for _, st := range prs.SkippedTasks {
+		new := v1.SkippedTask{}
+		st.convertTo(ctx, &new)
+		sink.SkippedTasks = append(sink.SkippedTasks, new)
+	}
+	sink.ChildReferences = nil
+	for _, cr := range prs.ChildReferences {
+		new := v1.ChildStatusReference{}
+		cr.convertTo(ctx, &new)
+		sink.ChildReferences = append(sink.ChildReferences, new)
+	}
+	sink.FinallyStartTime = prs.FinallyStartTime
+	if prs.Provenance != nil {
+		new := v1.Provenance{}
+		prs.Provenance.convertTo(ctx, &new)
+		sink.Provenance = &new
+	}
+	// If embedded-status is set to "both", both ChildReferences and TaskRuns/Runs
+	// will be populated. In this case, use the value from ChildReferences.
+	// TODO: to handle conversion when `embedded-status` feature flag is set to full or both as in #5813
```

Comment:
I think it might make sense to return an error for this conversion as well, because I'm not sure how the PipelineRun controller would behave in this scenario.

Reply:
This is indeed a concern once we are changing the reconciler when swapping the storage version. Noted in #5813  For now I suspect if this is necessary as `v1beta1` is still the storage version, so when converting up would be fine in all cases:
- Full: we are converting `taskruns` and `runs` into `childReferences`
- Minimal:  `childReferences` -> `childReferences`
- Both: `childReferences` with `taskruns` and `runs` are converted into `childReferences`, but the underneath logics here are mostly on `childReferences`
Therefore, there seems no information loss here when converting up. But I did add into the migration doc that this would require notifications for users at https://github.com/tektoncd/pipeline/pull/5827.

Reply:
So failing the conversion here might be limiting the usage of v1 pipelineRunStatus here if `status.taskRuns` and `status.runs` are not required for the conversion up here.

Reply:
In that case, you can probably remove the TODO since this logic doesn't need to change?

Reply:
@lbernick wanna check if my understanding is correct here:

Let say the storage version is `v1beta1`, an end user submits a `v1.PipelineRun`. The process that we handle it would be
1. The conversion webhook controller converts `v1` resource to `v1beta1` and store it in the etcd
2. The pipelinerun reconciler picks up the `v1beta1.Resource` (because it listens to the beta resource changes), and starts to reconcile it, store the updated `v1beta1` resource back to etcd
3. The end user wants to get the `v1.PipelineRun` resource back, the k8s api server receives the request, dispatch it to the conversion webhook, and conversion webhook convert the stored `v1beta1.PipelineRun` to `v1.PipelineRun`.

It this is how the request flow looks like, it seems like we don't have information loss from `v1beta1` to `v1` no matter what the feature flag `embedded-status`. Is that correct?

Reply:
Yeah that sounds right to me!

Reply:
Thanks for chiming in @XinruZhang for the clarifications!
And this TODO is removed @lbernick :)